### PR TITLE
docs(components): update method and parameter descriptions

### PR DESCRIPTION
**Pull request by brandyscarney**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe): 

## Other information

This will fix the following issue: https://github.com/ionic-team/ionic-docs/issues/634

This also fixes https://github.com/ionic-team/ionic-docs/issues/349, but it is still a legitimate bug in the docs generation and likely to happen again.

### CODE REVIEW

**Comment by brandyscarney** [comment]
I'm still working on reorder-group

File: core/src/components/action-sheet/action-sheet.tsx [hunk 1]
```diff
@@ -128,22 +128,25 @@ export class ActionSheet implements ComponentInterface, OverlayInterface {
 
   /**
    * Dismiss the action sheet overlay after it has been presented.
+   *
+   * @param data Any data to emit in the dismiss events.
+   * @param role The role of the element that is dismissing the action sheet. For example, 'cancel' or 'backdrop'.
```
Comment (liamdebeasi) [inline core/src/components/action-sheet/action-sheet.tsx]: Is there a set of accepted roles we can reference?

File: core/src/components/menu-controller/menu-controller.ts [hunk 2]
```diff
@@ -63,35 +75,44 @@ export class MenuController implements MenuControllerI {
   }
 
   /**
-   * Used to enable or disable a menu. For example, there could be multiple
-   * left menus, but only one of them should be able to be opened at the same
-   * time. If there are multiple menus on the same side, then enabling one menu
-   * will also automatically disable all the others that are on the same side.
+   * Enable or disable a menu. This is useful when there are multiple
+   * menus on the same side and only one of them should be allowed to
+   * open. Enabling a menu will automatically disable all other menus
```
Comment (liamdebeasi) [inline core/src/components/menu-controller/menu-controller.ts]: What does it mean for a menu to be disabled? Does it just mean it won't respond to swipe events or subsequent calls to `open`?

File: core/src/components/router/router.tsx [hunk 3]
```diff
@@ -88,6 +88,9 @@ export class Router implements ComponentInterface {
 
   /**
    * Navigate to the specified URL.
+   *
+   * @param url The url to navigate to.
+   * @param direction The direction of the animation.
```
Comment (liamdebeasi) [inline core/src/components/router/router.tsx]:
```suggestion
   * @param direction The direction of the animation. Defaults to `forward`.
```

File: core/src/components/slides/slides.tsx [hunk 4]
```diff
@@ -171,6 +174,10 @@ export class Slides implements ComponentInterface {
 
   /**
    * Transition to the specified slide.
+   *
+   * @param index The index of the slide to transition to.
+   * @param speed The transition duration (in ms).
+   * @param runCallbacks If true, the transition will produce transition events.
```
Comment (liamdebeasi) [inline core/src/components/slides/slides.tsx]: What is a "transition event"? Is that just an event that's fired when the transition finishes?

File: core/src/components/content/content.tsx [hunk 5]
```diff
@@ -199,15 +203,23 @@ export class Content implements ComponentInterface {
...
     return this.scrollToPoint(x + this.scrollEl.scrollLeft, y + this.scrollEl.scrollTop, duration);
   }
 
   /**
-   * Scroll to a specified X/Y location in the component
+   * Scroll to a specified X/Y location in the component.
+   *
+   * @param x The point to scroll to on the horizontal axis.
+   * @param y The point to scroll to on the vertical axis.
+   * @param duration The amount of time to take scrolling to that point.
```
Comment (liamdebeasi) [inline core/src/components/content/content.tsx]:
```suggestion
   * @param duration The amount of time to take scrolling to that point. Defaults to `0`.
```

File: core/src/components/action-sheet/action-sheet.tsx (hunk 1 above)
Comment (brandyscarney) [reply core/src/components/action-sheet/action-sheet.tsx]: It depends on the component, but all of the interfaces accept any string as well. Action sheet has the following:

```
role?: 'cancel' | 'destructive' | 'selected' | string;
```

Where alert is just `string`

**Pull request merged**